### DISCUSSION TITLE
DI UnitOfWorkManager事务，疑问

### DISCUSSION THREAD

**Discussion by langhaoabcd** [dotnetcore/FreeSql#842] [Q&A] [answered]
仓储事务，为何这里用继承后的Repository导致事务未生效，一定要使用父类的IBaseRepository<,> 事务才生效？

![image](https://user-images.githubusercontent.com/39892654/128378580-16f0e555-3513-4f8a-86ed-037b84a68081.png)

wiki：https://github.com/dotnetcore/FreeSql/wiki/DI-UnitOfWorkManager%E4%BA%8B%E5%8A%A1

**Comment by luoyunchong** [comment, marked as answer]
1、 [DefaultRepository.cs](https://github.com/dotnetcore/FreeSql/blob/master/FreeSql.DbContext/Repository/Repository/DefaultRepository.cs)中用
``  uowManger?.Binding(this);``绑定了事务。

2、[DependencyInjection.cs](https://github.com/dotnetcore/FreeSql/blob/master/FreeSql.DbContext/Repository/Extensions/DependencyInjection.cs)用
``AddFreeRepository``方法注入了泛型的仓储实现
```       
services.AddScoped(typeof(IBaseRepository<,>), typeof(DefaultRepository<,>));
```

如果 想重写，请直接继承现有的DefaultRepository,并传递UnitOfWorkManager或自己绑定下(``uowManger?.Binding(this);``)。

参考代码
[BlogRepository.cs](https://github.com/luoyunchong/dotnetcore-examples/blob/master/ORM/FreeSql/OvOv.FreeSql.AutoFac.DynamicProxy/Repositories/BlogRepository.cs)

```
    public class BlogRepository : DefaultRepository<Blog, int>, IBlogRepository
    {
        public BlogRepository(UnitOfWorkManager uowm) : base(uowm?.Orm, uowm)
        {
        }

        public List<Blog> GetBlogs()
        {
            return Select.Page(1, 10).ToList();
        }
    }
```

**Reply by langhaoabcd** [reply]
感谢，这是可行的
